### PR TITLE
sci-electronics/kicad and friends: 8.0.0 version bump

### PR DESCRIPTION
**Pull request by vowstar**
- sci-electronics/kicad: add 8.0.0
- sci-electronics/kicad-footprints: add 8.0.0
- sci-electronics/kicad-packages3d: add 8.0.0
- sci-electronics/kicad-symbols: add 8.0.0
- sci-electronics/kicad-templates: add 8.0.0
- app-doc/kicad-doc: add 8.0.0
- sci-electronics/kicad-meta: add 8.0.0
- {app-doc,sci-electronics}/kicad*: add me to maintainer

### CODE REVIEW

**Comment by dlan17** [comment]
for merging this PR, I'd like to at least have @zpuskas 's ACK, so let's wait for his review.. also note that @vowstar  want to help to co-maintain these packages (probably act as a backup maintainer)..

**Comment by vowstar** [comment]
Dear @dlan17,

Thank you for bringing these issues to my attention. I have reviewed the pkgcheck scan results and would like to address each point as follows:

- **Empty Global Assignment**: I have corrected the EmptyGlobalAssignment issue. This should resolve the problem and improve the overall code quality.

- **Redundant Uri Rename**: Regarding the ``RedundantUriRename`` warning, I have examined the previous code and understand your concern. The current implementation is designed to handle RC (Release Candidate) version numbers differently from non-RC versions. This approach, while it may trigger a redundant URI rename warning in some cases, ensures easier maintenance and better handling of different versioning scenarios. Given the trade-off between the warning and the practical benefits of the current approach, I suggest we maintain the existing code structure. However, I am open to alternative solutions if they can provide the same level of functionality without triggering the warning.

- **Failed REQUIRED_USE of Profile**: The issue regarding the ``default/linux/amd64/17.0/x32`` profile is indeed a bit more complex, as I lack direct experience with this particular scenario. From a maintenance perspective, it might be more practical to retain the current implementation, especially if the impact on overall functionality is minimal. However, if this issue could potentially lead to significant problems, it may warrant further investigation. Regarding the possibility of ignoring the ``x32`` profile specifically, I would appreciate your input or suggestions from anyone with more experience in this area.

In conclusion, my proposal is to rectify the ``EmptyGlobalAssignment`` issue, retain the current approach for ``RedundantUriRename`` for ease of maintenance, and seek further guidance on the ``REQUIRED_USE profile`` concern.

I look forward to your feedback and any further suggestions you might have.

**Comment by dlan17** [comment]
> Dear @dlan17,
> 
> Thank you for bringing these issues to my attention. I have reviewed the pkgcheck scan results and would like to address each point as follows:
> 
>     * **Empty Global Assignment**: I have corrected the EmptyGlobalAssignment issue. This should resolve the problem and improve the overall code quality.
> 

that's great!

>     * **Redundant Uri Rename**: Regarding the `RedundantUriRename` warning, I have examined the previous code and understand your concern. The current implementation is designed to handle RC (Release Candidate) version numbers differently from non-RC versions. This approach, while it may trigger a redundant URI rename warning in some cases, ensures easier maintenance and better handling of different versioning scenarios. Given the trade-off between the warning and the practical benefits of the current approach, I suggest we maintain the existing code structure. However, I am open to alternative solutions if they can provide the same level of functionality without triggering the warning.
> 

fine by me with current approach

>     * **Failed REQUIRED_USE of Profile**: The issue regarding the `default/linux/amd64/17.0/x32` profile is indeed a bit more complex, as I lack direct experience with this particular scenario. From a maintenance perspective, it might be more practical to retain the current implementation, especially if the impact on overall functionality is minimal. However, if this issue could potentially lead to significant problems, it may warrant further investigation. Regarding the possibility of ignoring the `x32` profile specifically, I would appreciate your input or suggestions from anyone with more experience in this area.
> 

probably first need to answer why there is "exactly one of L18N" restriction in ebuild {snip of code: ^^ ( ${LANG_USE} ) }, but if it's indeed the case, then I'm also fine with it..

> 
> In conclusion, my proposal is to rectify the `EmptyGlobalAssignment` issue, retain the current approach for `RedundantUriRename` for ease of maintenance, and seek further guidance on the `REQUIRED_USE profile` concern.
> 

thanks for taking care of this and the explanation..

> I look forward to your feedback and any further suggestions you might have.

File: sci-electronics/kicad/kicad-8.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		-DPYTHON_INCLUDE_DIR="$(python_get_includedir)"
+		-DPYTHON_LIBRARY="$(python_get_library_path)"
+
+		-DKICAD_INSTALL_DEMOS="$(usex examples)"
+		-DCMAKE_SKIP_RPATH="ON"
+
+		-DOCC_INCLUDE_DIR="${CASROOT}"/include/opencascade
+		-DOCC_LIBRARY_DIR="${CASROOT}"/$(get_libdir)/opencascade
+
+		-DKICAD_BUILD_QA_TESTS="$(usex test)"
```
Comment (zpuskas) [inline sci-electronics/kicad/kicad-8.0.0.ebuild:118]: We need to add:

`-DKICAD_SPICE_QA="$(usex test)"`

here.

Comment (zpuskas) [inline sci-electronics/kicad/kicad-8.0.0.ebuild:118]: New KiCad has telemetry code in it. While it's off by default we should explicitly disable it by just in case in the ebuild, but we also should give users the choice to enable it. For this we need to add the `telemetry` USE flag as well as the following code:

`-DKICAD_USE_SENTRY="$(usex telemetry)"`

**Review by zpuskas — CHANGES_REQUESTED** [review]
Thanks for contributing. Overall looks good, I'd just like to ask for some small improvements.

File: sci-electronics/kicad/kicad-8.0.0.ebuild (hunk 1 above)
Comment (vowstar) [reply sci-electronics/kicad/kicad-8.0.0.ebuild:121]: Done

Comment (vowstar) [reply sci-electronics/kicad/kicad-8.0.0.ebuild:121]: Done, thanks for review!

**Comment by zpuskas** [comment]
I've compiled KiCad-8.0.0 on my box and did a test run with some of my projects. LGTM.

**Review by zpuskas — APPROVED** [review]
